tests/unit/metrics/metrics_test.py

```diff
@@ -0,0 +1,162 @@
+import unittest
+
+from os import path
+
+from hazelcast.metrics import (
+    MetricsCompressor,
+    MetricDescriptor,
+    ProbeUnit,
+    _OutputBuffer,
+    _MetricsDictionary,
+)
+
+
+class MetricsCompatibilityTest(unittest.TestCase):
+    def test_compatibility(self):
+        here = path.abspath(path.dirname(__file__))
+        with open(path.join(here, "metrics.compatibility.binary"), "rb") as f:
+            expected_blob = f.read()
+
+        compressor = MetricsCompressor()
+
+        compressor.add_long(
+            MetricDescriptor(
+                metric="deltaMetric1",
+                prefix="prefix",
+                discriminator="ds",
+                discriminator_value="dsName1",
+                unit=ProbeUnit.COUNT,
+            ),
+            42,
+        )
+
+        compressor.add_double(
+            MetricDescriptor(
+                metric="deltaMetric2",
+                prefix="prefix",
+                discriminator="ds",
+                discriminator_value="dsName1",
+                unit=ProbeUnit.COUNT,
+            ),
+            -4.2,
+        )
+
+        compressor.add_long(
+            MetricDescriptor(
+                metric="longPrefixMetric",
+                prefix="a" * 254,
+                unit=ProbeUnit.BYTES,
+            ),
+            2147483647,
+        )
+
+        actual_blob = compressor.generate_blob()
+        self.assertEqual(expected_blob, actual_blob)
```

Comment:
nodejs test has a extractPayload method. Does python and java create exact same binary?

Reply:
Yes, they produce the same binary. I tested this on Linux and Windows as well